monai/fl/client/monai_algo.py

```diff
@@ -0,0 +1,214 @@
+
+
+def convert_global_weights(global_weights, local_var_dict):
+    """Helper function to convert global weights to local weights format"""
+    # Before loading weights, tensors might need to be reshaped to support HE for secure aggregation.
+    model_keys = global_weights.keys()
+    for var_name in local_var_dict:
+        if var_name in model_keys:
+            try:
+                # update the local dict
+                local_var_dict[var_name] = torch.as_tensor(global_weights[var_name])
+            except Exception as e:
+                raise ValueError(f"Convert weight from {var_name} failed.") from e
+    return local_var_dict
+
+
+class MonaiAlgo(ClientAlgo):
+    def __init__(
+        self,
+        config_train_file: Optional[str] = None,
+        config_predict_file: Optional[str] = None,
+        config_filters_file: Optional[str] = None,
+    ):
+        self.logger = logging.getLogger(self.__class__.__name__)
+        self.config_train_file = config_train_file
+        self.config_predict_file = config_predict_file
+        self.config_filters_file = config_filters_file
+
+        self.train_parser = None
+        self.predict_parser = None
+        self.filter_parser = None
+        self.trainer = None
+        self.predictor = None
+        self.pre_filters = None
+        self.post_weight_filters = None
+        self.post_predict_filters = None
+
+        self.phase = FlPhase.IDLE
+        self.client_name = None
+
+    def initialize(self, extra=None):
+        if extra is None:
+            extra = {}
+        self.logger.info(f"Initializing {self.client_name} ...")
+        self.client_name = extra.get(ExtraItems.CLIENT_NAME, "noname")
+        if self.config_train_file:
+            self.logger.info(f"Parsing trainer configuration from {self.config_train_file}")
+
+            self.train_parser = ConfigParser()
+            self.train_parser.read_config(self.config_train_file)
+            self.train_parser.parse()
+            self.trainer = self.train_parser.get_parsed_content("train#trainer")
+        if self.config_predict_file:
+            self.logger.info(f"Parsing predictor configuration from {self.config_predict_file}")
+
+            self.predict_parser = ConfigParser()
+            self.predict_parser.read_config(self.config_predict_file)
+            self.predict_parser.parse()
+            self.predictor = self.predict_parser.get_parsed_content("validate#evaluator")
+        if self.config_filters_file:
+            self.logger.info(f"Parsing filter configuration from {self.config_filters_file}")
+
+            self.filter_parser = ConfigParser()
+            self.filter_parser.read_config(self.config_filters_file)
+            self.filter_parser.parse()
+            # TODO: return default None in get_parsed_content() if key is not there, rather than using try/except
+            try:
+                self.pre_filters = self.filter_parser.get_parsed_content("pre_filters")
```

Comment:
If the idea here is to set `pre_filters` if "pre_filters" is present in the parsed file and do nothing otherwise, you can use `if "pre_filters" in filter_parser` to check instead. If you want to do nothing when an exception is actually raised this is fine but I would have some warning in the except block so it's not entirely silent. Unless there's good reason I think it's the intended practice to not check for `BaseException` in excepts but use `Exception`.

Reply:
Thanks. The try/except blocks have been replaced by using `get_parsed_content`'s default `ConfigItem`.

```
        self.pre_filters = self.parser.get_parsed_content(FiltersType.PRE_FILTERS, default=ConfigItem(None, FiltersType.PRE_FILTERS))
        self.post_weight_filters = self.parser.get_parsed_content(FiltersType.POST_WEIGHT_FILTERS, default=ConfigItem(None, FiltersType.POST_WEIGHT_FILTERS))
        self.post_evaluate_filters = self.parser.get_parsed_content(FiltersType.POST_EVALUATE_FILTERS, default=ConfigItem(None, FiltersType.POST_EVALUATE_FILTERS))
```